test/versioned/pino/pino.tap.js

```diff
@@ -0,0 +1,320 @@
+        )
+        tx.end()
+        t.equal(
+          agent.logs.getEvents().length,
+          1,
+          'should have log in aggregator after transaction ends'
+        )
+        t.same(agent.logs.getEvents(), [line], 'log should be same in aggregator')
+        t.end()
+      })
+    })
+
+    t.test('should properly handle child loggers', (t) => {
+      const childLogger = logger.child({ module: 'child' })
+      helper.runInTransaction(agent, 'pino-test', async (tx) => {
+        const level = 'info'
+        const message = 'My debug test'
+        logger[level](message)
+        const meta = agent.getLinkingMetadata()
+        const line = await once(stream, 'data')
+        t.validateAnnotations({ line, config, message, level })
+        t.equal(line['trace.id'], meta['trace.id'])
+        t.equal(line['span.id'], meta['span.id'])
+
+        const childMessage = 'this is a child message'
+        childLogger[level](childMessage)
+        const childLine = await once(stream, 'data')
+        t.validateAnnotations({ line: childLine, config, message: childMessage, level })
+        t.equal(childLine['trace.id'], meta['trace.id'])
+        t.equal(childLine['span.id'], meta['span.id'])
+        t.equal(
+          agent.logs.getEvents().length,
+          0,
+          'should have not have log in aggregator while transaction is active'
+        )
+        tx.end()
+        t.equal(
+          agent.logs.getEvents().length,
+          2,
+          'should have log in aggregator after transaction ends'
+        )
+        t.same(agent.logs.getEvents(), [childLine, line], 'log should be same in aggregator')
+        t.end()
+      })
+    })
+  })
+
+  t.test('forwarding + local decorating(favor forwarding)', async (t) => {
+    const config = setupAgent({
+      application_logging: {
+        enabled: true,
+        forwarding: { enabled: true },
+        local_decorating: { enabled: true },
+        metrics: { enabled: false }
+      }
+    })
+    const level = 'info'
+    const message = 'My debug test'
+    logger[level](message)
+    const meta = agent.getLinkingMetadata()
+    const line = await once(stream, 'data')
+    t.validateAnnotations({ line, config, message, level })
+    t.equal(line['trace.id'], meta['trace.id'])
+    t.equal(line['span.id'], meta['span.id'])
+    t.equal(
+      agent.logs.getEvents().length,
+      1,
+      'should have log in aggregator after transaction ends'
+    )
+    t.same(agent.logs.getEvents(), [line], 'log should be same in aggregator')
+    t.end()
+  })
+
+  t.test('metrics', (t) => {
+    t.autoend()
+
+    t.test('should count logger metrics', (t) => {
+      setupAgent({
+        application_logging: {
+          enabled: true,
+          local_decorating: { enabled: false },
+          forwarding: { enabled: false },
+          metrics: { enabled: true }
+        }
+      })
+      helper.runInTransaction(agent, 'pino-test', async () => {
```

Comment:
Oh hey, I recognise this test. Copied over from the pino decorator, correct?

Reply:
yes. i made sure all relevant tests were ported over